### DISCUSSION TITLE
How to build an Azure VM for MacOS agent

### DISCUSSION THREAD

**Discussion by ghost** [actions/virtual-environments#8974] [General questions] [answered]
Hello, I'd like to create a VM on Azure using the MacOS image and the Packer scripts available on the [Actions runner-images repository](https://github.com/actions/runner-images/tree/main). However, I can't find the instructions to build the VM image for MacOS; I only see instructions for the Windows and Linux operating systems.

In the script helpers/GenerateResourcesAndImage.ps1, the MacOS ImageType enum is missing. 

enum ImageType {
    Windows2019   = 1
    Windows2022   = 2
    Ubuntu2004    = 3
    Ubuntu2204    = 4
    UbuntuMinimal = 5
}

Could you please suggest any reference documentation for building a MacOS image VM on Azure?

**Comment by ilia-shipitsin** [comment, marked as answer]
Hello, @dcupellini-dev

Unfortunately Azure does not offer MacOS based VMs. Because of this, it will be difficult to find any official documentation that will be useful to you in this particular case.
